packages/server-wallet/jest/with-peers-setup-teardown.ts

```diff
@@ -61,96 +58,133 @@ export const participantIdB = 'b';
-    try {
-      await Promise.all([DBAdmin.dropDatabase(aEngineConfig), DBAdmin.dropDatabase(bEngineConfig)]);
-
-      await Promise.all([
-        DBAdmin.createDatabase(aEngineConfig),
-        DBAdmin.createDatabase(bEngineConfig),
-      ]);
-
-      await Promise.all([
-        DBAdmin.migrateDatabase(aEngineConfig),
-        DBAdmin.migrateDatabase(bEngineConfig),
-      ]);
-
-      peerEngines = {
-        a: await Engine.create(aEngineConfig),
-        b: await Engine.create(bEngineConfig),
-      };
-
-      if (withWalletSeeding) {
-        await seedAlicesSigningWallet(peerEngines.a.knex);
-        await seedBobsSigningWallet(peerEngines.b.knex);
-      }
-
-      participantA = {
-        signingAddress: await peerEngines.a.getSigningAddress(),
-        participantId: participantIdA,
-        destination: makeDestination(
-          '0x00000000000000000000000000000000000000000000000000000000000aaaa1'
-        ),
-      };
-      participantB = {
-        signingAddress: await peerEngines.b.getSigningAddress(),
-        participantId: participantIdB,
-        destination: makeDestination(
-          '0x00000000000000000000000000000000000000000000000000000000000bbbb2'
-        ),
-      };
-
-      const participantEngines = [
-        {participantId: participantIdA, engine: peerEngines.a},
-        {participantId: participantIdB, engine: peerEngines.b},
-      ];
-
-      const handler = createTestMessageHandler(participantEngines, peerEngines.a.logger);
-      messageService = (await TestMessageService.create(
-        handler,
-        peerEngines.a.logger
-      )) as TestMessageService;
-    } catch (error) {
-      logger.error(error, 'getPeersSetup failed');
-      throw error;
+export async function getPeersSetup(withWalletSeeding = false): Promise<PeerSetup> {
+  try {
+    await Promise.all([DBAdmin.dropDatabase(aEngineConfig), DBAdmin.dropDatabase(bEngineConfig)]);
+
+    await Promise.all([
+      DBAdmin.createDatabase(aEngineConfig),
+      DBAdmin.createDatabase(bEngineConfig),
+    ]);
+
+    await Promise.all([
+      DBAdmin.migrateDatabase(aEngineConfig),
+      DBAdmin.migrateDatabase(bEngineConfig),
+    ]);
+
+    const peerEngines = {
+      a: await Engine.create(aEngineConfig),
+      b: await Engine.create(bEngineConfig),
+    };
+
+    if (withWalletSeeding) {
+      await seedAlicesSigningWallet(peerEngines.a.knex);
+      await seedBobsSigningWallet(peerEngines.b.knex);
     }
-  };
+
+    const participantA = {
+      signingAddress: await peerEngines.a.getSigningAddress(),
+      participantId: participantIdA,
+      destination: makeDestination(
+        '0x00000000000000000000000000000000000000000000000000000000000aaaa1'
+      ),
+    };
+    const participantB = {
+      signingAddress: await peerEngines.b.getSigningAddress(),
+      participantId: participantIdB,
+      destination: makeDestination(
+        '0x00000000000000000000000000000000000000000000000000000000000bbbb2'
+      ),
+    };
```

Comment:
This almost-repeated block (see lines 85-98) is a bit of a smell. Maybe the most basic fix is to throw

``` ts
const destinationA: Destination = makeDestination(
  '0x00000000000000000000000000000000000000000000000000000000000aaaa1'
);
const destinationB: Destination = makeDestination(
  '0x00000000000000000000000000000000000000000000000000000000000bbbb2'
);
```
into the top of this file and handing them into the destination field here.

Reply:
Done in dc937e7